### PR TITLE
 Preparation for 0.7.2 release

### PR DESCRIPTION
**Pull request by RMeli**

### CODE REVIEW

File: RELEASE_PROCEDURE.md [hunk 1]
```diff
@@ -38,6 +38,8 @@ DLA-Future follows [Semantic Versioning](https://semver.org).
    with local repository. Exclude blocks delimited by `###` comments. These are only intended for the
    internal spack package.
 
+1. For patch releases: update `CHANGELOG` of `master` branch.
+
```
Comment (msimberg) [inline RELEASE_PROCEDURE.md:42]: I was thinking this could be "merge `version_X.Y` branch into master"`? It doesn't always work if we plan on ever patching old versions after we've released a new minor/major version, but will work in 99% of cases I would expect. You can maybe separate this change into a different PR to not block on this discussion?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## DLA-Future 0.7.2
+
+### Changes
+
+- Changed C ScaLAPACK API indexing convention to 1-based for partial eigenspectrum. (#1248)
```
Comment (msimberg) [inline CHANGELOG.md:7]: Is this a bugfix? I'm fine either way and won't matter too much especially if we mark 0.7.0 and 0.7.1 deprecated in spack, but by making this release 0.7.2 and not 0.8.0 we're signaling that this is a bugfix.

**Review by msimberg — APPROVED** [review]
Happy as it is, but added a couple of comments that may need discussion.

File: CHANGELOG.md (hunk 2 above)
Comment (RMeli) [reply CHANGELOG.md:7]: I wondered the same, but I went with "changed" because it was not technically a bug. Happy to mark it as a bugfix if preferred.

Comment (msimberg) [reply CHANGELOG.md:7]: It's not a strong preference from me. Mostly wondering out loud...

File: RELEASE_PROCEDURE.md (hunk 1 above)
Comment (RMeli) [reply RELEASE_PROCEDURE.md]: Yes, that is fair. https://github.com/eth-cscs/DLA-Future/pull/1250

**Review by msimberg — APPROVED** [review]
